publisher.py

```diff
@@ -367,7 +381,15 @@ def process_images(self):
                 for tag in image.release_tags:
                     repository = "%s/%s" % (target, image.repository.name)
                     self.tag_image(image, repository, tag)
-                    self.publish_image(target, repository, tag)
+                    retry_count = 1
+                    while retry_count <= self.retry_limit:
+                        self.log.info("Pushing %s:%s to %s (%d/%d)", repository, tag, target, retry_count, self.retry_limit)
+                        try:
+                            return self.publish_image(target, repository, tag)
```

Comment:
If I read it properly, `publish_image` for `dry_run=True` will return `None`, which will propagate down and print "Error occurred" in main. Not a serious bug, but you can change this.

Reply:
:+1: